src/main/java/eu/arrowhead/client/library/ArrowheadService.java

```diff
@@ -448,6 +457,133 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
+	 */
+	public WebSocketConnectionManager connnectServiceWS(final WebSocketHandler handler, final String address, final int port, final String serviceUri, final String interfaceName,
+			final String token, final String... queryParams) {
+
+		if (Utilities.isEmpty(address)) {
+			throw new InvalidParameterException("address cannot be null or blank.");
+		}
+		if (Utilities.isEmpty(serviceUri)) {
+			throw new InvalidParameterException("serviceUri cannot be null or blank.");
+		}
+		if (Utilities.isEmpty(interfaceName)) {
+			throw new InvalidParameterException("interfaceName cannot be null or blank.");
+		}
+
+		String[] validatedQueryParams;
+		if (queryParams == null) {
+			validatedQueryParams = new String[0];
+		} else {
+			validatedQueryParams = queryParams;
+		}
+
+		UriComponents uri;
+		if(!Utilities.isEmpty(token)) {
+			final List<String> query = new ArrayList<>();
+			query.addAll(Arrays.asList(validatedQueryParams));
+			query.add(CommonConstants.REQUEST_PARAM_TOKEN);
+			query.add(token);
+			uri = Utilities.createURI("ws", address, port, serviceUri, query.toArray(new String[query.size()]));
+		} else {
+			uri = Utilities.createURI("ws", address, port, serviceUri, validatedQueryParams);
+		}
+
+		WebSocketConnectionManager manager = new WebSocketConnectionManager(new StandardWebSocketClient(), handler, uri.toString() );
+		return manager;
+	}
+
+
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Make WSS connection with the specified service reachability details.
+	 *
+	 * @param trustStore The trusted certificates
+	 * @param keyStore The certificate
+	 * @param password The passwords
+	 * @param httpMethod HttpMethod enum which represents the method how the service is available.
+	 * @param address String value which represents the host where the service is available.
+	 * @param port int value which represents the port where the service is available
+	 * @param serviceUri String value which represents the URI where the service is available.
+	 * @param interfaceName String value which represents the name of the interface used for the communication. Usable interfaces could be received in orchestration response.
+	 * @param token (nullable) String value which represents the token for being authorized at the provider side if necessary. Token could be received in orchestration response per interface type.
+	 * @param payload (nullable) Object type which represents the required payload of the http(s) request if any necessary.
+	 * @param queryParams (nullable) String... variable arguments which represent the additional key-value http(s) query parameters if any necessary. E.g.: "k1", "v1", "k2", "v2".
+	 * @return the response received from the provider
+	 *
+	 * @throws InvalidParameterException when service URL can't be assembled.
+	 * @throws AuthException when ssl context or access control related issue happened.
```

Comment:
Not see where it whould throw AuthException

Reply:
Fixed